### ISSUE TITLE
Searching for 'ssi' causes a server error

### ISSUE THREAD

**Issue by gvwilson** [swcarpentry/amy#257]
1. Go to http://amy.software-carpentry.org:8080/workshops/search
2. Enter 'ssi' (without quotes) in the search box.
3. Search.
4. Boom.

Search works for 'toronto'.  Maybe 'ssi' is returning too many results?

**Comment by pbanaszkiewicz** [comment]
Reproduced with production database. The error is:

```
Environment:


Request Method: POST
Request URL: http://127.0.0.1:8000/workshops/search

Django Version: 1.7.3
Python Version: 3.4.3
Installed Applications:
('django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'workshops',
 'django.contrib.admin',
 'crispy_forms')
Installed Middleware:
('django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware')


Template error:
In template /home/piotr/workspace/SwC/amy/workshops/templates/workshops/search.html, error at line 26
   Reverse for 'event_details' with arguments '('',)' and keyword arguments '{}' not found. 1 pattern(s) tried: ['workshops/event/(?P<event_ident>[\\w-]+)/?$']
   16 :   {% for s in sites %}


   17 :   <li><a class="searchresult" href="{% url 'site_details' s.domain %}">{{ s.domain }}</a></li>


   18 :   {% endfor %}


   19 : </ul>


   20 : {% endif %}


   21 : 


   22 : <h2>Events</h2>


   23 : {% if events %}


   24 : <ul>


   25 :   {% for e in events %}


   26 :   <li><a class="searchresult" href=" {% url 'event_details' e.slug %} ">{{ e.slug }}</a></li>


   27 :   {% endfor %}


   28 : </ul>


   29 : {% endif %}


   30 : 


   31 : <h2>Persons</h2>


   32 : {% if persons %}


   33 : <ul>


   34 :   {% for p in persons %}


   35 :   <li><a class="searchresult" href="{% url 'person_details' p.id %}">{{ p }}</a></li>


   36 :   {% endfor %}


Traceback:
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/core/handlers/base.py" in get_response
  111.                     response = wrapped_callback(request, *callback_args, **callback_kwargs)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/contrib/auth/decorators.py" in _wrapped_view
  21.                 return view_func(request, *args, **kwargs)
File "/home/piotr/workspace/SwC/amy/workshops/views.py" in search
  655.     return render(request, 'workshops/search.html', context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/shortcuts.py" in render
  50.     return HttpResponse(loader.render_to_string(*args, **kwargs),
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/loader.py" in render_to_string
  178.         return t.render(context_instance)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/base.py" in render
  148.             return self._render(context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/base.py" in _render
  142.         return self.nodelist.render(context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/base.py" in render
  844.                 bit = self.render_node(node, context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/debug.py" in render_node
  80.             return node.render(context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/loader_tags.py" in render
  126.         return compiled_parent._render(context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/base.py" in _render
  142.         return self.nodelist.render(context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/base.py" in render
  844.                 bit = self.render_node(node, context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/debug.py" in render_node
  80.             return node.render(context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/loader_tags.py" in render
  126.         return compiled_parent._render(context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/base.py" in _render
  142.         return self.nodelist.render(context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/base.py" in render
  844.                 bit = self.render_node(node, context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/debug.py" in render_node
  80.             return node.render(context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/loader_tags.py" in render
  126.         return compiled_parent._render(context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/base.py" in _render
  142.         return self.nodelist.render(context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/base.py" in render
  844.                 bit = self.render_node(node, context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/debug.py" in render_node
  80.             return node.render(context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/loader_tags.py" in render
  65.                 result = block.nodelist.render(context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/base.py" in render
  844.                 bit = self.render_node(node, context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/debug.py" in render_node
  80.             return node.render(context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/defaulttags.py" in render
  312.                 return nodelist.render(context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/base.py" in render
  844.                 bit = self.render_node(node, context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/debug.py" in render_node
  80.             return node.render(context)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/defaulttags.py" in render
  201.                             nodelist.append(node.render(context))
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/defaulttags.py" in render
  458.                         six.reraise(*exc_info)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/utils/six.py" in reraise
  659.         raise value
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/template/defaulttags.py" in render
  444.             url = reverse(view_name, args=args, kwargs=kwargs, current_app=context.current_app)
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/core/urlresolvers.py" in reverse
  551.     return iri_to_uri(resolver._reverse_with_prefix(view, prefix, *args, **kwargs))
File "/home/piotr/workspace/anaconda/envs/amy3k/lib/python3.4/site-packages/django/core/urlresolvers.py" in _reverse_with_prefix
  468.                              (lookup_view_s, args, kwargs, len(patterns), patterns))

Exception Type: NoReverseMatch at /workshops/search
Exception Value: Reverse for 'event_details' with arguments '('',)' and keyword arguments '{}' not found. 1 pattern(s) tried: ['workshops/event/(?P<event_ident>[\\w-]+)/?$']
```

**Comment by wking** [comment]
On Wed, May 06, 2015 at 01:00:24PM -0700, Piotr Banaszkiewicz wrote:
> Exception Value: Reverse for 'event_details' with arguments '('',)'
> and keyword arguments '{}' not found. 1 pattern(s) tried:
> ['workshops/event/(?P<event_ident>[\\w-]+)/?$']

Looks like there's an Event where Event.slug is the empty string?

**Comment by pbanaszkiewicz** [comment]
@wking precisely. Fixed in #258.
